### PR TITLE
Mark lock file as invalid when see parsing errors

### PR DESCRIPTION
**Pull request by ChengTian**
parent https://github.com/aspnet/dnx/issues/1371

### CODE REVIEW

File: src/Microsoft.Framework.Runtime/DependencyManagement/LockFileFormat.cs [hunk 1]
```diff
@@ -71,7 +82,7 @@ private LockFile ReadLockFile(JObject cursor)
         {
             var lockFile = new LockFile();
             lockFile.Islocked = ReadBool(cursor, "locked", defaultValue: false);
-            lockFile.Version = ReadInt(cursor, "version", defaultValue: int.MaxValue);
+            lockFile.Version = ReadInt(cursor, "version", defaultValue: int.MinValue);
```
Comment (ChengTian) [inline src/Microsoft.Framework.Runtime/DependencyManagement/LockFileFormat.cs]: Version int.MinValue is always out-of-date. We use this way to mark a lockfile as invalid.

Comment (davidfowl) [inline src/Microsoft.Framework.Runtime/DependencyManagement/LockFileFormat.cs]: Why is the parameter defaultValue instead of minimumValue?

Comment (ChengTian) [inline src/Microsoft.Framework.Runtime/DependencyManagement/LockFileFormat.cs]: `defaultValue` means this value should be used when there is no `version` property. Here we want to use `int.MinValue` when `version` property doesn't appear in lockfile.

**Comment by davidfowl** [comment]
:shipit:

**Pull request merged**